WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -34,89 +35,154 @@ private class WeeklyRoundupDataProvider {
+        completion: @escaping (Result<SiteStats?, Error>) -> Void
+    ) {
+        guard let periodEndDate = getPeriodEndDate() else {
+            DDLogError("Something's wrong with the period end date selection.")
             return
         }
 
-        var blogStats = [Blog: StatsSummaryData]()
-        var statsProcessed = 0
+        let group = DispatchGroup()
+        var blogStats = SiteStats()
 
         for site in sites {
             guard let authToken = site.account?.authToken else {
                 continue
             }
 
-            let wpApi = WordPressComRestApi.defaultApi(oAuthToken: authToken, userAgent: WPUserAgent.wordPress())
-
-            guard let dotComID = site.dotComID?.intValue else {
-                onError(DataRequestError.dotComSiteWithoutDotComID(site))
-                continue
-            }
-
-            let statsServiceRemote = StatsServiceRemoteV2(wordPressComRestApi: wpApi, siteID: dotComID, siteTimezone: site.timeZone)
+            group.enter()
 
-            statsServiceRemote.getData(for: .week, endingOn: periodEndDate, limit: 1) { (timeStats: StatsSummaryTimeIntervalData?, error) in
+            self.fetchStats(for: site, endingOn: periodEndDate, authToken: authToken, in: context) { [weak self] result in
                 defer {
-                    statsProcessed = statsProcessed + 1
-
-                    if statsProcessed == sites.count {
-                        let bestBlogStats = self.filterBest(5, from: blogStats)
-
-                        completion(.success(bestBlogStats))
+                    group.leave()
+                }
+                guard let self else {
+                    return
+                }
+                switch result {
+                case .failure(let error):
+                    self.onError(error)
+                case.success(let stats):
+                    if let stats {
+                        blogStats[site.objectID] = stats
                     }
                 }
+            }
+        }
+
+        group.notify(queue: .global()) {
+            context.perform {
+                let bestBlogStats = self.filterBest(5, from: blogStats)
+                completion(.success(bestBlogStats))
+            }
+        }
+    }
 
+    /// Fetches weekly stats data for a given site.
+    ///
+    /// This function fetches the stats for a single site and passes the result to a completion handler.
+    /// If it encounters any error during fetching, it calls the completion handler with an appropriate error object.
+    /// The completion handler is executed in the same queue as the provided `NSManagedObjectContext`.
+    ///
+    /// **Note:** The completion handler is executed on the context's queue.
+    private func fetchStats(
+        for site: Blog,
+        endingOn periodEndDate: Date,
+        authToken: String,
+        in context: NSManagedObjectContext,
+        completion: @escaping (Result< StatsSummaryData?, Error>) -> Void
```

Comment:
There is an extra space after `Result< `.

Reply:
Resolved in fac2f074ab1fb0a9832ef449b0fe2ee37df8d99c